public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -156,6 +158,11 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
       this.annotations = {
         QueryEditor: FluxQueryEditor,
       };
+    } else {
```

Comment:
Could we ever have more then one query editor?

Reply:
Within `this.annotations`? Just room for one QueryEditor

https://github.com/grafana/grafana/blob/6fe2d4724a23512ebdf43e9b04ec93363bd89095/packages/grafana-data/src/types/annotations.ts#L115

Reply:
There is a case for refactoring this in the future. We have the Flux query editor which is used in the QueryEditor and for Flux Annotations. We also have the InfluxQL query editor which is different than the InfluxQL annotations editor.